src/migration/network.rs

```diff
@@ -2,21 +2,108 @@ use std::net::IpAddr;
 
 use anyhow::{anyhow, Context, Result};
 use giganto_client::ingest::network::{
-    Bootp, Conn, DceRpc, Dhcp, Dns, Ftp, FtpCommand, Http, Kerberos, Ldap, Mqtt, Nfs, Ntlm, Radius,
-    Rdp, Smb, Smtp, Ssh, Tls,
+    Bootp, Conn, DceRpc, Dhcp, Dns, Ftp, FtpCommand, Http, Kerberos, Ldap, MalformedDns, Mqtt, Nfs,
+    Ntlm, Radius, Rdp, Smb, Smtp, Ssh, Tls,
 };
 
 use super::{
     parse_comma_separated, parse_giganto_timestamp, parse_post_body, TryFromGigantoRecord,
 };
 
+fn parse_hex_body(field: &str) -> Result<Vec<Vec<u8>>> {
+    let trimmed = field.trim();
+    if trimmed.is_empty() || trimmed == "-" || trimmed == "[]" {
+        return Ok(Vec::new());
+    }
+    if !trimmed.starts_with('[') || !trimmed.ends_with(']') {
+        return Err(anyhow!("invalid hex body: {field}"));
+    }
+    let inner = &trimmed[1..trimmed.len() - 1];
+    if inner.trim().is_empty() {
+        return Ok(Vec::new());
+    }
+
+    if !inner.bytes().any(|b| b == b'[') {
```

Comment:
@pott-cho 
Since the `field` is always in the form of `[[..]]`, after `let inner = &trimmed[1..trimmed.len() - 1];`, `inner` should always be in the form of `[..]`. So it seems like the if condition would never be true. Is there any case I might be overlooking? If not, I believe this if statement should be removed.